src/pages/index.js

```diff
@@ -36,17 +103,29 @@ const cardSection = new Section(
   },
   cardList
 );
-cardSection.renderItems(initialCards);
 
+////////////////////////
 // ----- POPUPS ----- //
+////////////////////////
 
 const userInformation = new UserInfo({
   nameSelector: ".profile__title",
   descriptionSelector: ".profile__description",
 });
 
 const editProfilePopup = new PopupWithForm("#modal__edit-profile", (data) => {
-  userInformation.setUserInfo(data);
+  editProfilePopup.renderLoadingMessage("Saving...");
+  api.editUserInformation(data).then((data) => {
+    return new Promise((resolve) => {
+      userInformation.setUserInfo(data);
+      resolve();
+    })
+      .then(() => {
+        editProfilePopup.close();
+        editProfilePopup.resetSaveButton("Save");
```

Comment:
Change button state in `finally` block, otherwise the button text won’t change back if an error occurred

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Fix that everywhere